src/cobra/flux_analysis/deletion.py

```diff
@@ -405,3 +405,47 @@ def double_gene_deletion(
         processes=processes,
         **kwargs
     )
+
+
+@pd.api.extensions.register_dataframe_accessor("knockout")
+@pd.api.extensions.register_series_accessor("knockout")
+class KnockoutAccessor:
+    """Access unique combinations of reactions in deletion results."""
+
+    def __init__(self, pandas_obj):
+        """Set up the accessor.
+
+        Parameters:
+        -----------
+        pandas_obj : pd.DataFrame or pd.Series
+            A result from one of the deletion methods.
+        """
+        self._validate(pandas_obj)
+        self._result = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        # verify it is a deletion results
+        if obj.index.name != "ids":
+            raise AttributeError("Must be DataFrame returned by a deletion method.")
+
+    def __getitem__(self, args):
+        """Return the deletion result for a particular set of knocked entities.
+
+        Parameters:
+        -----------
+        args : iterable of str, cobra.Reaction or cobra.Gene
+
+        Returns:
+        --------
+        pd.DataFrame
+            The deletion result where the chosen entities have been deleted.
+        """
+        if not any(isinstance(args, t) for t in [list, tuple, set]):
+            args = [args]
+
+        if isinstance(args[0], Reaction) or isinstance(args[0], Gene):
+            args = tuple(sorted(set(obj.id for obj in args)))
```

Comment:
It'd be weird to pass duplicates when the typical use-case is double knock-outs but I guess you're right to be careful.

```suggestion
            args = tuple(sorted({obj.id for obj in args}))
```

Reply:
May happen if you want the diagonal elements like `rxn1, rxn1` which get indexed as `(rxn1, )` in the deletion module.